### ISSUE TITLE
Change default open behaviour to "open as stack"

### ISSUE THREAD

**Issue by jni** [napari/napari#2603]
## 🚀 Feature

Currently when dragging a folder or group of files to napari, the default is to open them as layers. This is a problem because napari scales *very well* with stacking, while it scales extremely poorly with layers. I have found myself very often in the situation in which I drag a folder full of tiffs to napari and freeze the process. On the other hand, if the reverse happens (I get a stack but wanted layers), it's very fast to undo — delete one layer and try again. In the future it can be even faster — split axis into layers.

**Comment by sofroniewn** [comment]
Fine with me. Will be very nice when we have a "stack to layers" option in the context menu!

**Comment by GenevieveBuckley** [comment]
> Fine with me. Will be very nice when we have a "stack to layers" option in the context menu!

Linking https://github.com/napari/napari/pull/1322 - we'd need a menu button to call these functions. This seems like it'd be another good first issue.

**Comment by tlambert03** [comment]
> Currently when dragging a folder the default is to open them as layers...

I'm a little confused by this.  I don't get that behavior.   with a folder, I get a stack

```python
import os
import tempfile
import numpy as np
import tifffile
import napari

with tempfile.TemporaryDirectory() as tmpdir:
    for i in range(100):
        fname = os.path.join(tmpdir, f'{i:04d}.tif')
        print(f'writing {fname}')
        data = np.random.randint(0, 65000, (1024, 1024)).astype("uint16")
        tifffile.imsave(fname, data)

    viewer = napari.Viewer()
    print("opening...")
    viewer.open(tmpdir, stack=False)
    print("done...")

    napari.run()
```

(same thing if I save the files, and drag the folder onto the viewer)
I assume I'm missing something in the chain there... @jni?

**Comment by tlambert03** [comment]
> Currently when dragging a folder or group of files to napari, the default is to open them as layers.

This isn't entirely true actually.  The default with a single folder is still dictated by `utils.io.magic_imread` which defaults to `stack=True` ... it might be confusing that `viewer.open`  has `stack=False` but the docstring explains the difference:

```
If a list of strings is passed and ``stack`` is ``True``, then the
entire list will be passed to plugins.  It is then up to individual
plugins to know how to handle a list of paths.  If ``stack`` is
``False``, then the ``path`` list is broken up and passed to plugin
readers one by one.  by default False.
```

A single folder is not a list of strings and so opens as a stack (if handled by `magic_imread`).  I'm not immediately sold on the idea that if you've dragged multiple _individual_ files that they should be implicitly stacked:

https://user-images.githubusercontent.com/1609449/117520898-c6bacf80-af78-11eb-8305-5c1d840667ea.mov

**Comment by tlambert03** [comment]
> I'm not immediately sold on the idea that if you've dragged multiple individual files that they should be implicitly stacked:

(but if the list paths provided to `viewer.open()` is longer than some threshold, we might be able to make some reasonable crash-preventing switch?)

**Comment by tlambert03** [comment]
@jni can you comment on these last couple posts?  it's unclear to me that this is actually an issue

**Comment by jni** [comment]
Yeah, two things:

- folder was wrong: that was the ome-zarr plugin intepreting axis 1 of non-ome zarrs as channels. :facepalm: 
- lists of files is still an issue — `napari *.tif` would happily try to open umpteen layers, and I come across this with some regularity.

**Comment by tlambert03** [comment]
the `*` use case from the CLI makes a lot of sense... but I still don't know that I'd want a few files opened in the GUI to automatically stack.  What do you think about the n_layers threshold idea?  or, making stacking behavior default for CLI, but not for gui?

**Comment by sofroniewn** [comment]
> A single folder is not a list of strings and so opens as a stack (if handled by magic_imread). I'm not immediately sold on the idea that if you've dragged multiple individual files that they should be implicitly stacked:

Yeah I tend to agree with @tlambert03 here, I like individual folder "stacked", list of individual files "not stacked". 

> the * use case from the CLI makes a lot of sense

But this is also a good point!

I'm open to `n_layers threshold`, but don't love it, seems fairly arbitrary - maybe `>5` people do have 5 channel data pretty frequently, but I'd say above that is less common

> making stacking behavior default for CLI, but not for gui?

This is pretty reasonable too, can you pass "channel_axis" kwarg to CLI? that might help
